### ISSUE TITLE
zig fetch errors with sha256 git repositories

### ISSUE THREAD

**Issue by GalaxyShard** [ziglang/zig#21888]
### Zig Version

0.14.0-dev.2097+d30e28754

### Steps to Reproduce and Observed Behavior

Repro:

```sh
$ zig fetch --save git+https://codeberg.org/GalaxyShard/zig-nds-bindings.git
error: unable to iterate refs: EndOfStream
```

Or alternatively:

```sh
$ zig build
/home/galaxyshard/projects/abcbuild.zig.zon:14:20: error: unable to iterate refs: EndOfStream
            .url = "git+https://codeberg.org/GalaxyShard/zig-nds-bindings.git#297a06aab678414a350cc9bb68b5705b9969913601397c065dc847e1610d2200",
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

(I believe there is also supposed to be a path seperator between the project name and build.zig.zon)

### Expected Behavior

The repository should be fetched and checked out without error, whether it uses sha256 or sha1.

**Comment by liinyun** [comment]
I have the same problem in fedora 40,  zig version = 0.13.0

**Comment by ianprime0509** [comment]
The current Git package fetching code only supports SHA-1 repositories, but it should be reasonably straightforward to expand it to support SHA-256: the underlying [pack format](https://git-scm.com/docs/pack-format) is the same (except with all SHA-1 hashes replaced with SHA-256 hashes), and the [Git protocol](https://git-scm.com/docs/protocol-v2#_object_format) allows servers to indicate which hash format(s) they support via the `object-format` capability. I hope to try implementing this soon.

**Comment by ianprime0509** [comment]
The main part of this issue is now covered by #22222, which I just opened. The other part (missing path separator between the project name and `build.zig.zon`) was fixed by #22032.
